test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/ModelBuildingHelpers.cs

```diff
@@ -176,6 +176,33 @@ public static EdmModel GetModelWithIllegalActionOverloads()
             return model;
         }
 
+        public static IEdmModel GetModelWithOperationOverloadsWithSameName()
+        {
+            EdmEntityType vegetableType = new EdmEntityType("Test", "Vegetable", null, true, false);
+            IEdmStructuralProperty id = vegetableType.AddStructuralProperty("ID", EdmCoreModel.Instance.GetInt32(false));
+            vegetableType.AddKeys(id);
+            EdmCollectionTypeReference collectionTypeReference = new EdmCollectionTypeReference(new EdmCollectionType(new EdmEntityTypeReference(vegetableType, false)));
+
+            EdmEntityContainer container = new EdmEntityContainer("Test", "Container");
+            var set = container.AddEntitySet("Vegetables", vegetableType);
+
+            var function1 = new EdmFunction("Test", "Function", new EdmEntityTypeReference(vegetableType, true), false /*isBound*/, null /*entitySetPath*/, false);
+            function1.AddParameter("p1", collectionTypeReference);
+            function1.AddParameter("p2", EdmCoreModel.Instance.GetInt32(false));
+
+            var function2 = new EdmFunction("Test", "Function", new EdmEntityTypeReference(vegetableType, true), false /*isBound*/, null /*entitySetPath*/, false);
+            function2.AddParameter("p1", new EdmEntityTypeReference(vegetableType, true));
+            function2.AddParameter("p2", EdmCoreModel.Instance.GetInt32(false));

```

Comment:
Does this pass model validation? It should not -- you shouldn't be able to have two overloads of the same unbound function with the same parameter names.

Reply:
Yes. I will follow up on that. In the meanwhile, it is a valid scenario for action overloads which differ by the binding type and there the parameter name does not even matter. Hypothetically, can the parameter name be the same but differ in type for a bound action.

To be on the safe side, I changed the above test to be a valid scenario with a different name. I will follow up with the behavior in model validation separately.